src/core/socket.h

```diff
@@ -164,6 +168,21 @@ struct Socket {
         RateLimit trigger_limit;
 };
 
+struct SocketPeer {
+        unsigned n_ref;
+
+        Socket *socket;
+
+        int family;
+        union sockaddr_union peer;
+};
+
+SocketPeer *socket_peer_ref(SocketPeer *p);
+SocketPeer *socket_peer_unref(SocketPeer *p);
+
+DEFINE_TRIVIAL_CLEANUP_FUNC(SocketPeer*, socket_peer_unref);
+#define _cleanup_link_unref_ _cleanup_(socket_peer_unrefp)
```

Comment:
we try to avoid such type specific cleanup macro definitions these days, except for the most common of types. Hence, please use use _cleanup_(socket_peer_unrefp) everywhere, and avoid defining the _cleanup_link_unref_ macro as a shortcut for that.